stdlib/LinearAlgebra/src/uniformscaling.jl

```diff
@@ -210,6 +210,7 @@ end
     rmul!(inv(A), J.λ)
 \(J::UniformScaling, A::AbstractVecOrMat) = J.λ == 0 ? throw(SingularException(1)) : J.λ\A
 \(A::AbstractMatrix, J::UniformScaling) = rmul!(inv(A), J.λ)
+\(F::Factorization, J::UniformScaling) = F \ J(size(F,1))
```

Comment:
This seems to allocate an intermediate `Diagonal` matrix object. Why not simply
```julia
\(F::Factorization, J::UniformScaling) = F \ J.λ
```
similar to two lines above?

Reply:
Because there aren't methods to deal with an `Adjoint{<:Factorization}`:
```julia
julia> X = randn(3,2)
3×2 Array{Float64,2}:
 -0.457525   -1.20586
 -1.07724     1.25765
  0.0835881  -1.55446

julia> J = randn()*I
UniformScaling{Float64}
1.3897141915176197*I

julia> qr(X) \ J.λ
ERROR: MethodError: no method matching adjoint(::LinearAlgebra.QRCompactWY{Float64,Array{Float64,2}})
Closest candidates are:
  adjoint(::Missing) at missing.jl:100
  adjoint(::Number) at number.jl:169
  adjoint(::Adjoint) at /home/justin/devel/ext/julia/usr/share/julia/stdlib/v1.4/LinearAlgebra/src/adjtrans.jl:152
  ...
Stacktrace:
 [1] \(::LinearAlgebra.QRCompactWY{Float64,Array{Float64,2}}, ::Float64) at ./operators.jl:563
 [2] top-level scope at REPL[9]:1

julia> lu(X) \ J.λ
ERROR: MethodError: no method matching /(::Float64, ::Adjoint{Float64,LU{Float64,Array{Float64,2}}})
Closest candidates are:
  /(::Float64, ::Float64) at float.jl:407
  /(::R, ::S) where {R<:Real, S<:Complex} at complex.jl:332
  /(::Union{Float16, Float32, Float64}, ::BigFloat) at mpfr.jl:457
  ...
Stacktrace:
 [1] \(::LU{Float64,Array{Float64,2}}, ::Float64) at ./operators.jl:563
 [2] top-level scope at REPL[10]:1
```
I thought about trying to specialize further, but I decided to go for the simplest change to add the feature at all, and a separate PR could add performance improvements. (Mostly because I don't really have a mental model of how all the linear algebra and factorization code paths work yet.)

Reply:
I see. I often mess up the slashes and where the numbers go.